libraries/iio/iio.c

```diff
@@ -866,11 +867,12 @@ ssize_t iio_register(struct iio_desc *desc, struct iio_device *dev_descriptor,
 
 	desc->xml_desc = aux;
 	/* Print the new device xml at the end of the xml */
-	iio_generate_device_xml(iio_interface->dev_descriptor, iio_interface->name,
+	iio_generate_device_xml(iio_interface->dev_descriptor,
+				iio_interface->name,
 				desc->dev_count,
 				desc->xml_desc + desc->xml_size_to_last_dev,
 				new_size - desc->xml_size_to_last_dev);
-
+	sprintf((char *)iio_interface->dev_id, "device%d", (int)desc->dev_count);
```

Comment:
If the device IDs are zero based dev_count here could end up being one more than we want it to. Is that the case?

Reply:
Will not be one more. If you register one device will have the id: "device0" and after desc->dev_count will increment to 1 if this is what you mean.